cpp/ql/src/semmle/code/cpp/models/interfaces/FunctionInputsAndOutputs.qll

```diff
@@ -35,73 +117,178 @@ class InParameter extends FunctionInput, TInParameter {
+   *   type `float`) on return from the function.
+   * There is no `FunctionOutput` for which `isParameterDeref(0)` holds, because `n` is neither a
+   *   pointer nor a reference.
+   */
+  predicate isParameterDeref(ParameterIndex i) { none() }
+
+  /**
+   * Holds if this is the output value pointed to by a pointer parameter to a function, or the
+   *   output value referred to by a reference parameter to a function, where the parameter has
+   *   index `index`.
+   * DEPRECATED: Use `isParameterDeref(index)` instead.
+   */
+  deprecated final predicate isOutParameterPointer(ParameterIndex index) { isParameterDeref(index) }
+
+  /**
+   * Holds if this is the output value pointed to by the `this` pointer of an instance member
+   *   function.
+   * ```
+   * struct C {
+   *   void mfunc(int n, char* p, float& r);
+   * };
+   * ```
+   * `isQualifierObject()` holds for the `FunctionOutput` that represents the value of `*this` (with
+   *   type `C`) on return from the function.
+   */
+  predicate isQualifierObject() { none() }
+
+  /**
+   * Holds if this is the output value pointed to by the `this` pointer of an instance member
+   *   function.
+   * DEPRECATED: Use `isQualifierObject()` instead.
+   */
+  deprecated final predicate isOutQualifier() { isQualifierObject() }
+
+  /**
+   * Holds if this is the value returned by a function.
+   * Example:
+   * ```
+   * int getInt();
+   * char* getPointer();
+   * float& getReference();
+   * ```
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the value returned by
+   *   `getInt()` (with type `int`).
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the value returned by
+   *   `getPointer()` (with type `char*`).
+   * `isReturnValue()` holds for the `FunctionOutput` that represents the "value" of the reference
+   *   returned by `getReference()` (with type `float&`), _not_ the value of the referred-to
+   *   `float`.
+   */
+  predicate isReturnValue() { none() }
+
+  /**
+   * Holds if this is the value returned by a function.
+   * DEPRECATED: Use `isReturnValue()` instead.
+   */
+  deprecated final predicate isOutReturnValue() { isReturnValue() }
+
+  /**
+   * Holds if this is the output value pointed to by the return value of a function, if the function
+   *   returns a pointer, or the output value referred to by the return value of a function, if the
+   *   function returns a reference.
+   * Example:
+   * ```
+   * char* getPointer();
+   * float& getReference();
+   * int getInt();
+   * ```
+   * `isReturnValueDeref()` holds for the `FunctionOutput` that represents the value of
+   *   `*getPointer()` (with type `char`).
+   * `isReturnValueDeref()` holds for the `FunctionOutput` that represents the value of
+   *   `getReference()` (with type `float`).
+   * There is no `FunctionOutput` of `getInt()` for which `isReturnValueDeref()` holds because the
+   *   return type of `getInt()` is neither a pointer nor a reference.
```

Comment:
Outside of triple backticks, line breaks are not significant in QLDoc. I think this would read better as a bullet list so all three items don't get run together in a single paragraph. If you prefer them to be run together, I think it's better to reflect that in the source text by running them together there as well.

Reply:
Fixed.